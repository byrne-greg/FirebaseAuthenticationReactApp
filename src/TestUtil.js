import React from "react";
import { MemoryRouter } from "react-router-dom";

const withTestRouter = Component => <MemoryRouter>{Component}</MemoryRouter>;

export { withTestRouter };
