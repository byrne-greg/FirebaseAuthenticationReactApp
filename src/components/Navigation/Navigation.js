import React, { useContext } from "react";
import { Link } from "react-router-dom";

import routes from "../../constants/Routes";
import { SignOutButton } from "../SignOut";
import { AuthUserContext } from "../Session";

const Navigation = () => {
  const authUser = useContext(AuthUserContext);
  return authUser ? <NavigationAuth /> : <NavigationPublic />;
};

const NavigationAuth = () => (
  <div>
    <ul>
      <li>
        <Link to={routes.LANDING.url}>{routes.LANDING.title}</Link>
      </li>
      <li>
        <Link to={routes.HOME.url}>{routes.HOME.title}</Link>
      </li>
      <li>
        <Link to={routes.ACCOUNT.url}>{routes.ACCOUNT.title}</Link>
      </li>
      {/* <li>
        <Link to={routes.ADMIN.url}>Admin</Link>
      </li> */}
      <li>
        <SignOutButton />
      </li>
    </ul>
  </div>
);

const NavigationPublic = () => (
  <div>
    <ul>
      <li>
        <Link to={routes.LANDING.url}>{routes.LANDING.title}</Link>
      </li>
      <li>
        <Link to={routes.SIGN_IN.url}>{routes.SIGN_IN.title}</Link>
      </li>
      <li>
        <Link to={routes.SIGN_UP.url}>{routes.SIGN_UP.title}</Link>
      </li>
    </ul>
  </div>
);

export default Navigation;
export { NavigationAuth, NavigationPublic };
