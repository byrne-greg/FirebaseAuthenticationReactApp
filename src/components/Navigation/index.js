import Navigation, { NavigationAuth, NavigationPublic } from "./Navigation";

export default Navigation;
export { NavigationAuth, NavigationPublic };
