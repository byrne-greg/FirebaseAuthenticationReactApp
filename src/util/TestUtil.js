import React from "react";
import { MemoryRouter } from "react-router-dom";
import { FirebaseContext } from "../components/Firebase";
import { AuthUserContext } from "../components/Session";

const withMockFirebase = (mockFirebase, Component) => (
  <FirebaseContext.Provider value={mockFirebase}>
    {Component}
  </FirebaseContext.Provider>
);

const withMockAuthUserContext = (value, Component) => (
  <AuthUserContext.Provider value={value}>{Component}</AuthUserContext.Provider>
);

const withTestRouter = Component => <MemoryRouter>{Component}</MemoryRouter>;

export { withTestRouter, withMockFirebase, withMockAuthUserContext };
