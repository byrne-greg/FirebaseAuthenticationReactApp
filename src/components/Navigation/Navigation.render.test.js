import React from "react";
import { render, cleanup } from "@testing-library/react";
import { withTestRouter, withMockAuthUserContext } from "../../util/TestUtil";
import Navigation from ".";

afterEach(cleanup);

describe("Navigation render tests", () => {
  it("when user is unauthenticated, the public navigation bar renders as expected", () => {
    // GIVEN

    // WHEN
    const page = render(withTestRouter(<Navigation />));

    // THEN
    expect(page.baseElement).toMatchSnapshot();
  });

  it("when user is authenticated, the authenticated user navigation bar renders as expected", () => {
    // GIVEN

    // WHEN
    const page = render(
      withTestRouter(
        withMockAuthUserContext({ mockUser: true }, <Navigation />)
      )
    );

    // THEN
    expect(page.baseElement).toMatchSnapshot();
  });
});
