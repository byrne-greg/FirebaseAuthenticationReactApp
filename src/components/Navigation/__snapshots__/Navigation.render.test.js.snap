// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navigation render tests when user is authenticated, the authenticated user navigation bar renders as expected 1`] = `
<body>
  <div>
    <div>
      <ul>
        <li>
          <a
            href="/"
          >
            Landing
          </a>
        </li>
        <li>
          <a
            href="/home"
          >
            Home
          </a>
        </li>
        <li>
          <a
            href="/account"
          >
            Account
          </a>
        </li>
        <li>
          <button
            id="sign-out-button"
            type="button"
          >
            Sign Out
          </button>
        </li>
      </ul>
    </div>
  </div>
</body>
`;

exports[`Navigation render tests when user is unauthenticated, the public navigation bar renders as expected 1`] = `
<body>
  <div>
    <div>
      <ul>
        <li>
          <a
            href="/"
          >
            Landing
          </a>
        </li>
        <li>
          <a
            href="/signin"
          >
            Sign In
          </a>
        </li>
        <li>
          <a
            href="/signup"
          >
            Sign Up
          </a>
        </li>
      </ul>
    </div>
  </div>
</body>
`;
